File: src/main/java/org/apache/fineract/client/services/MeetingsApi.java
/**
 * Licensed to the Apache Software Foundation (ASF) under one
 * or more contributor license agreements. See the NOTICE file
 * distributed with this work for additional information
 * regarding copyright ownership. The ASF licenses this file
 * to you under the Apache License, Version 2.0 (the
 * "License"); you may not use this file except in compliance
 * with the License. You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied. See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */
package org.apache.fineract.client.services;

import org.apache.fineract.client.CollectionFormats.*;

import io.reactivex.Observable;
import io.reactivex.Completable;
import retrofit2.http.*;

import okhttp3.RequestBody;
import okhttp3.ResponseBody;
import okhttp3.MultipartBody;


import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

public interface MeetingsApi {
  /**
   * 
   * 
   * @param entityType  (required)
   * @param entityId  (required)
   * @param body  (optional)
   * @return Observable&lt;String&gt;
   */
  @Headers({
    "Content-Type:application/json"
  })
  @POST("{entityType}/{entityId}/meetings")
  Observable<String> createMeeting(
    @retrofit2.http.Path("entityType") String entityType, @retrofit2.http.Path("entityId") Long entityId, @retrofit2.http.Body String body
  );

  /**
   * 
   * 
   * @param entityType  (required)
   * @param entityId  (required)
   * @param meetingId  (required)
   * @return Observable&lt;String&gt;
   */
  @DELETE("{entityType}/{entityId}/meetings/{meetingId}")
  Observable<String> deleteMeeting(
    @retrofit2.http.Path("entityType") String entityType, @retrofit2.http.Path("entityId") Long entityId, @retrofit2.http.Path("meetingId") Long meetingId
  );

  /**
   * 
   * 
   * @param entityType  (required)
   * @param entityId  (required)
   * @param meetingId  (required)
   * @param command  (optional)
   * @param body  (optional)
   * @return Observable&lt;String&gt;
   */
  @Headers({
    "Content-Type:application/json"
  })
  @POST("{entityType}/{entityId}/meetings/{meetingId}")
  Observable<String> performMeetingCommands(
    @retrofit2.http.Path("entityType") String entityType, @retrofit2.http.Path("entityId") Long entityId, @retrofit2.http.Path("meetingId") Long meetingId, @retrofit2.http.Query("command") String command, @retrofit2.http.Body String body
  );

  /**
   * 
   * 
   * @param meetingId  (required)
   * @param entityType  (required)
   * @param entityId  (required)
   * @return Observable&lt;String&gt;
   */
  @GET("{entityType}/{entityId}/meetings/{meetingId}")
  Observable<String> retrieveMeeting(
    @retrofit2.http.Path("meetingId") Long meetingId, @retrofit2.http.Path("entityType") String entityType, @retrofit2.http.Path("entityId") Long entityId
  );

  /**
   * 
   * 
   * @param entityType  (required)
   * @param entityId  (required)
   * @param limit  (optional)
   * @return Observable&lt;String&gt;
   */
  @GET("{entityType}/{entityId}/meetings")
  Observable<String> retrieveMeetings(
    @retrofit2.http.Path("entityType") String entityType, @retrofit2.http.Path("entityId") Long entityId, @retrofit2.http.Query("limit") Integer limit
  );

  /**
   * 
   * 
   * @param entityType  (required)
   * @param entityId  (required)
   * @param calendarId  (optional)
   * @return Observable&lt;String&gt;
   */
  @GET("{entityType}/{entityId}/meetings/template")
  Observable<String> template11(
    @retrofit2.http.Path("entityType") String entityType, @retrofit2.http.Path("entityId") Long entityId, @retrofit2.http.Query("calendarId") Long calendarId
  );

  /**
   * 
   * 
   * @param entityType  (required)
   * @param entityId  (required)
   * @param meetingId  (required)
   * @param body  (optional)
   * @return Observable&lt;String&gt;
   */
  @Headers({
    "Content-Type:application/json"
  })
  @PUT("{entityType}/{entityId}/meetings/{meetingId}")
  Observable<String> updateMeeting(
    @retrofit2.http.Path("entityType") String entityType, @retrofit2.http.Path("entityId") Long entityId, @retrofit2.http.Path("meetingId") Long meetingId, @retrofit2.http.Body String body
  );

}

File: src/main/kotlin/org/openapitools/client/apis/MeetingsApi.kt
package org.openapitools.client.apis

import retrofit2.http.Body
import retrofit2.http.DELETE
import retrofit2.http.GET
import retrofit2.http.POST
import retrofit2.http.PUT
import retrofit2.http.Path
import retrofit2.http.Query


interface MeetingsApi {
    /**
     *
     *
     * Responses:
     *  - 0: default response
     *
     * @param entityType
     * @param entityId
     * @param body  (optional)
     * @return [kotlin.String]
     */
    @POST("v1/{entityType}/{entityId}/meetings")
    suspend fun createMeeting(
        @Path("entityType") entityType: String,
        @Path("entityId") entityId: Long,
        @Body body: String? = null
    ): String

    /**
     *
     *
     * Responses:
     *  - 0: default response
     *
     * @param entityType
     * @param entityId
     * @param meetingId
     * @return [kotlin.String]
     */
    @DELETE("v1/{entityType}/{entityId}/meetings/{meetingId}")
    suspend fun deleteMeeting(
        @Path("entityType") entityType: String,
        @Path("entityId") entityId: Long,
        @Path("meetingId") meetingId: Long
    ): String

    /**
     *
     *
     * Responses:
     *  - 0: default response
     *
     * @param entityType
     * @param entityId
     * @param meetingId
     * @param command  (optional)
     * @param body  (optional)
     * @return [kotlin.String]
     */
    @POST("v1/{entityType}/{entityId}/meetings/{meetingId}")
    suspend fun performMeetingCommands(
        @Path("entityType") entityType: String,
        @Path("entityId") entityId: Long,
        @Path("meetingId") meetingId: Long,
        @Query("command") command: String? = null,
        @Body body: String? = null
    ): String

    /**
     *
     *
     * Responses:
     *  - 0: default response
     *
     * @param meetingId
     * @param entityType
     * @param entityId
     * @return [kotlin.String]
     */
    @GET("v1/{entityType}/{entityId}/meetings/{meetingId}")
    suspend fun retrieveMeeting(
        @Path("meetingId") meetingId: Long,
        @Path("entityType") entityType: String,
        @Path("entityId") entityId: Long
    ): String

    /**
     *
     *
     * Responses:
     *  - 0: default response
     *
     * @param entityType
     * @param entityId
     * @param limit  (optional)
     * @return [kotlin.String]
     */
    @GET("v1/{entityType}/{entityId}/meetings")
    suspend fun retrieveMeetings(
        @Path("entityType") entityType: String,
        @Path("entityId") entityId: Long,
        @Query("limit") limit: Int? = null
    ): String

    /**
     *
     *
     * Responses:
     *  - 0: default response
     *
     * @param entityType
     * @param entityId
     * @param calendarId  (optional)
     * @return [kotlin.String]
     */
    @GET("v1/{entityType}/{entityId}/meetings/template")
    suspend fun template11(
        @Path("entityType") entityType: String,
        @Path("entityId") entityId: Long,
        @Query("calendarId") calendarId: Long? = null
    ): String

    /**
     *
     *
     * Responses:
     *  - 0: default response
     *
     * @param entityType
     * @param entityId
     * @param meetingId
     * @param body  (optional)
     * @return [kotlin.String]
     */
    @PUT("v1/{entityType}/{entityId}/meetings/{meetingId}")
    suspend fun updateMeeting(
        @Path("entityType") entityType: String,
        @Path("entityId") entityId: Long,
        @Path("meetingId") meetingId: Long,
        @Body body: String? = null
    ): String

}
